src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/EETypeCreator.cs

```diff
@@ -1058,6 +1058,13 @@ public static RuntimeTypeHandle CreateEEType(TypeDesc type, TypeBuilderState sta
                 requireVtableSlotMapping = true;
                 pTemplateEEType = null;
             }
+#if !PROJECTN
+            else if (type.IsSzArray && ((ArrayType)type).ElementType.IsPointer)
+            {
+                pTemplateEEType = typeof(void*[]).TypeHandle.ToEETypePtr();
```

Comment:
I think that you could simply have this use the same template as the multidimensional array path, and then we wouldn't need all the #if !PROJECTN bits.